lib/elix/responders/gif_me.ex

```diff
@@ -0,0 +1,47 @@
+defmodule Elix.Responders.GifMe do
+  @moduledoc """
+  Searches the Giphy public API and return the URL
+  of a GIF matching the search term.
+  """
+
+  use Hedwig.Responder
+
+  @base_url "http://api.giphy.com/v1/gifs/search"
+  @api_key "dc6zaTOxFJmzC" # Giphy’s public beta API key
+  @sample_size 10
+
+  @usage """
+  gif me <term> - Replies with a GIF URL matching the term
+  """
+  hear ~r/gif me (.+)/i, msg do
+    reply(msg, get_gif_url(msg.matches[0]))
+  end
+
+  defp get_gif_url(search_term) do
+    search_term
+    |> build_api_url
+    |> make_request
+    |> get_random_image_url
+  end
```

Comment:
This doesn't do any error handling. I'd like to use Elixir's `with` construct to handle errors in `make_request` (e.g. timeouts) and `get_random_image_url` (e.g. a different hash structure).